ballerina/provider.bal

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/ai;
+import ballerina/lang.regexp;
+import ballerina/uuid;
+import ballerinax/mistral;
+
+const DEFAULT_MISTRAL_AI_SERVICE_URL = "https://api.mistral.ai/v1";
+const DEFAULT_MAX_TOKEN_COUNT = 512;
+const DEFAULT_TEMPERATURE = 0.7d;
```

Comment:
Why we set this as a non-zero value?

Reply:
We have used 0.7 since is a `balanced` temperature.